### DISCUSSION TITLE
Benchmarks

### DISCUSSION THREAD

**Discussion by e3dio** [uNetworking/uWebSockets.js#692] [General]
![image](https://user-images.githubusercontent.com/85405955/152336356-b444af63-ad34-441c-8428-ecab26a5a4c5.png)

**Comment by ghost** [comment]
Quite sad how Node.js absolutely chokes at 270k events per second regardless of what those events are.

**Comment by Maximusya** [comment]
@e3dio care to add more "infamous" things to your nice pictures?
- default socket.io
- socket.io-with-uWebSockets.js https://socket.io/blog/socket-io-4-4-0/#support-for-uwebsocketjs

Thank you.

**Reply by e3dio** [reply]
It depends on the specific case, there are cases where Socket.io is 50x slower than uWS.js if you are publishing quickly between different topics with sockets subscribed to multiple topics, same with ws library. It does that by grouping messages by socket before sending them out. Using uWS.js as backend for Socket.io helps but ideally I'd just use uWS.js, trying to work with Socket.io was annoying setting up benchmark, did not like it

**Reply by e3dio** [reply]
1000 clients subscribed to 20 topics each, publish to 20 topics each iteration

![Slide-1-copy-3](https://user-images.githubusercontent.com/85405955/153069437-37745a4b-e67e-40e7-9a71-93fd9ceb86e5.jpg)
